### PR TITLE
fix bug where front end display the wrong calender year courses

### PR DESCRIPTION
**Pull request by haowei99**
Also added more details when referencing links to the academic programs: 

Please try to test this with refresh - and make sure the links still show up to the desire years

Try out the links in the program selection bar as well

image

### CODE REVIEW

**Review by kevinfwl — APPROVED** [review]
LGTM, just some small nitpicks above.

File: src/store/modules/courseSelection.js [hunk 1]
```diff
@@ -837,7 +837,8 @@ const actions = {
           option:
             getters.specRequirements.length !== 0
               ? getters.specRequirements[0].info.program_name
-              : ""
+              : "",
+          calender_year: getters.majorRequirements[0].info.year
```
[the remark this answers is not in the document]
Comment (haowei99) [reply src/store/modules/courseSelection.js]: sounds good, I will do that

File: src/components/CourseSelectionPage/ProgramSelectionBar.vue [hunk 2]
```diff
@@ -51,18 +51,42 @@ export default {
...
+  },
   methods: {
     goToContactPage() {
       this.$router.push("/Contact");
     },
     goToAboutUsPage() {
       this.$router.push("/About");
-    }
+    },
+    getLinkToUnderGradCalender(link) {
```
[the remark this answers is not in the document]
Comment (haowei99) [reply src/components/CourseSelectionPage/ProgramSelectionBar.vue]: done - take a look

**Review by kevinfwl — APPROVED** [review]
LGTM!

**Pull request merged**